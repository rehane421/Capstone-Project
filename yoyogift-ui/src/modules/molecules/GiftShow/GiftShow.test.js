import React from "react";
import { shallow } from "../../../enzyme";
import { shallowToJson } from "enzyme-to-json";
import { GiftShow } from "./GiftShow";

describe("GiftShow", () => {
  let wrapper, props;

  props = {
    data: {
      cardComments: [
        {
          first_name: "ashish",
          last_name: "roshan",
          comment: "Nice discount",
          commented_on: "02/28/2020"
        }
      ]
    },
    getEmail: jest.fn()
  };
  it("should render GiftShow", () => {
    wrapper = shallow(<GiftShow {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    console.log(shallowToJson(wrapper));
  });
});
