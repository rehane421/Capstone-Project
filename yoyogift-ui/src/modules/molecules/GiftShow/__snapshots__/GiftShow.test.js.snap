// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftShow should render GiftShow 1`] = `
<div
  style={
    Object {
      "padding": "4%",
    }
  }
>
  <div
    className="detailsPage"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={0}
    >
      <WithStyles(ForwardRef(Grid))
        className="cardContainer"
        item={true}
        sm={5}
        xs={12}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={0}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={6}
            xs={6}
          >
            <img
              alt="vendorImage"
              className="vendorImage"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            sm={6}
            xs={6}
          >
            <p
              className="cardName"
            />
            <span
              style={
                Object {
                  "marginLeft": "30%",
                }
              }
            >
              <label
                className="label"
              >
                Points:
              </label>
            </span>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <DraggableDialog
          getEmail={[Function]}
        />
        <p>
          <label
            className="label"
          >
            NUMBER OF CARDS AVAILABLE:
             
          </label>
        </p>
        <p>
          <label
            className="label"
          >
            EXPIRY DATE: 
          </label>
          NaN-NaN-NaN
        </p>
        <div>
          <p
            className="label"
          >
             RATING & REVIEWS:
          </p>
          <div
            className="commentsSection"
          >
            <p
              className="commentorName"
            >
               
              ashish
               
              roshan
              <span
                className="commentDate"
              >
                2020-02-28
              </span>
            </p>
            <p
              className="comment"
            >
              Nice discount
            </p>
            <StarRatingComponent
              editing={false}
              emptyStarColor="#333"
              name="rating"
              starColor="#ffb400"
              starCount={5}
            />
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="detailsContainer"
        item={true}
        sm={6}
        xs={12}
      >
        <p
          className="vendorName"
        />
        <p
          className="cardDescription"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;
